nixos/modules/programs/fish.nix

```diff
@@ -169,18 +178,77 @@ in
       end
     '';
 
+    programs.fish.interactiveShellInit = ''
+      # add completions generated by NixOS to $fish_complete_path
+      begin
+        set -l joined (string join " " $fish_complete_path)
+        set -l prev_joined (string replace --regex "[^\s]*generated_completions.*" "" $joined)
+        set -l post_joined (string replace $prev_joined "" $joined)
+        set -l prev (string split " " (string trim $prev_joined))
+        set -l post (string split " " (string trim $post_joined))
+        set fish_complete_path $prev "/etc/fish/generated_completions" $post
+      end
+    '';
+
+    environment.etc."fish/generated_completions".source =
+      let
+        # paths later in the list will overwrite those already linked
+        destructiveSymlinkJoin =
```

Comment:
Any reason not to use buildEnv here?

Reply:
Excellent point - I didn't know about buildEnv. I've switched to use that instead.